var/spack/repos/builtin/packages/strumpack/package.py

```diff
@@ -148,44 +148,55 @@ def cmake_args(self):
                 args.append('-DHIP_HIPCC_FLAGS=--amdgpu-target={0}'.
                             format(",".join(rocm_archs)))
 
-        if self.spec.satisfies('@:5.1.1'):
-            self.test_data_dir = 'examples/data'
-        else:
-            self.test_data_dir = 'examples/sparse/data'
-        self.test_src_dir = 'test'
         return args
 
+    test_src_dir = 'test'
+
+    @property
+    def test_data_dir(self):
+        """Return the stand-alone test data directory."""
+        add_sparse = not self.spec.satisfies('@:5.1.1')
+        return join_path('examples', 'sparse' if add_sparse else '', 'data')
+
     @run_after('install')
     def cache_test_sources(self):
         """Copy the example source files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
         self.cache_extra_test_sources([self.test_data_dir, self.test_src_dir])
 
     def _test_example(self, test_prog, test_dir, test_cmd, test_args):
-        tmpbld_dir = '{0}/_BUILD'.format(test_dir)
-        with working_dir(tmpbld_dir, create=True):
-            with open('{0}/CMakeLists.txt'.format(tmpbld_dir), 'w') as mkfile:
-                mkfile.write('cmake_minimum_required(VERSION 3.13)\n')
-                mkfile.write('project(StrumpackSmokeTest LANGUAGES CXX)\n')
-                mkfile.write('find_package(STRUMPACK REQUIRED)\n')
-                mkfile.write('add_executable({0} ../{0}.cpp)\n'.
-                             format(test_prog))
-                mkfile.write('target_link_libraries({0} '.format(test_prog) +
-                             'PRIVATE STRUMPACK::strumpack)\n')
-
-            opts = self.std_cmake_args
-            opts += self.cmake_args()
-            opts += ['.']
-            self.run_test('cmake', opts, [], installed=False, work_dir='.')
-            self.run_test('make')
-            with set_env(OMP_NUM_THREADS='4'):
-                self.run_test(test_cmd, test_args, installed=False,
-                              purpose='test: strumpack smoke test',
-                              skip_missing=False, work_dir='.')
-        self.run_test('rm', ['-fR', tmpbld_dir])
+        if not self.spec.satisfies('+openmp+parmetis'):
+            print('Skipping {0} test: +openmp+parmetis required'.
+                  format(test_prog))
+            return
+
+        with open('{0}/CMakeLists.txt'.format(test_dir), 'w') as mkfile:
+            mkfile.write('cmake_minimum_required(VERSION 3.13)\n')
+            mkfile.write('project(StrumpackSmokeTest LANGUAGES CXX)\n')
+            mkfile.write('find_package(STRUMPACK REQUIRED)\n')
+            mkfile.write('find_package(METIS REQUIRED)\n')
```

Comment:
Is this necessary?
I think this should be included automatically through `find_package(STRUMPACK REQUIRED)`

Reply:
Thanks for pointing that out.  I'd forgotten to remove that line before pushing the fix.

Reply:
Removed